uma/protocol.py

```diff
@@ -445,11 +454,38 @@ class PubkeyResponse(JSONable):
 	It can be safely cached until this expiration (or forever if null).
     """
 
+    def get_signing_pubkey(self) -> bytes:
+        # the first cert in the chain is the leaf (sender's) cert
+        if self.signing_cert_chain and self.signing_cert_chain[0]:
+            return get_pubkey(self.signing_cert_chain[0])
+        if self.signing_pubkey:
+            return self.signing_pubkey
+        raise InvalidRequestException("Signing pubkey is required for uma.")
+
+    def get_encryption_pubkey(self) -> bytes:
+        # the first cert in the chain is the leaf (sender's) cert
+        if self.encryption_cert_chain and self.encryption_cert_chain[0]:
+            return get_pubkey(self.encryption_cert_chain[0])
+        if self.encryption_pubkey:
+            return self.encryption_pubkey
+        raise InvalidRequestException("Encryption pubkey is required for uma.")
+
     def to_dict(self) -> Dict[str, Any]:
-        json_dict: Dict[str, Any] = {
-            "signingPubKey": self.signing_pubkey.hex(),
-            "encryptionPubKey": self.encryption_pubkey.hex(),
-        }
+        json_dict: Dict[str, Any] = {}
+        if self.signing_cert_chain:
+            json_dict["signingCertChain"] = [
+                cert.public_bytes(encoding=serialization.Encoding.PEM).hex()
+                for cert in self.signing_cert_chain
+            ]
+        if self.encryption_cert_chain:
+            json_dict["encryptionCertChain"] = [
+                cert.public_bytes(encoding=serialization.Encoding.PEM).hex()
+                for cert in self.encryption_cert_chain
+            ]
+        if self.signing_pubkey:
+            json_dict["signingPubKey"] = self.signing_pubkey.hex()
```

Comment:
I think if the raw keys aren't present, but the certs are, maybe we should just also encode the `signingPubKey` and `encryptionPubKey` as the pubkeys extracted from certs. That way things keep working with v0 counter-parties.

Reply:
hm I was envisioning vasps would call `create_pubkey_response` here https://github.com/uma-universal-money-address/uma-python-sdk/pull/29/files#diff-2af4efff0d5f35dffc167eab72e193020ea72731f84f2d90b67ccae36f561aacR77 instead of the constructor directly which does that

Reply:
Ohh I see 🤔. Any reason we couldn't do both? I'm wondering if we can make this as idiot-proof as possible if callers use the constructor directly. I could definitely see people not realizing that the `create_pubkey_response` function exists since this is a pretty simple object.

Reply:
no reason!! that makes sense, will update!